packages/jest/src/matchers.tsx

```diff
@@ -4,84 +4,74 @@ const kebabCase = (str: string) =>
     .replace(/\s+/g, '-')
     .toLowerCase();
 
+const mapProperties = (properties: Record<string, any>) =>
+  Object.keys(properties).map(property => `${kebabCase(property)}:${properties[property]}`);
+
+const getMountedProperties = () =>
+  Array.from(document.styleSheets)
+    .map(sheet =>
+      // @ts-ignore
+      sheet.cssRules.map((rule: CSSRule) => rule.style.cssText)
+    )
+    .join(' ');
+
 export function toHaveCompiledCss(
   this: jest.MatcherUtils,
   element: HTMLElement,
   ...args: [{ [key: string]: string } | string, string]
 ): jest.CustomMatcherResult {
   const [property, value] = args;
   const properties = typeof property === 'string' ? { [property]: value } : property;
-  let styleElement = element.parentElement && element.parentElement.querySelector('style');
-
-  if (!styleElement) {
-    // There wasn't a style element found within - let's check the head for it instead.
-    const styleElements = Array.from(document.head.querySelectorAll('style'));
-    for (const tag of styleElements) {
-      if (tag.innerHTML.includes(element.className)) {
-        styleElement = tag as HTMLStyleElement;
-        break;
-      }
-    }
-  }
+  const inlineStyleTag = element.parentElement && element.parentElement.querySelector('style');
```

Comment:
is this correct? there could be multiple style tags tied to a singe element (like the composed StyleDiv test above)

Reply:
Hey, im might be wrong here but i thought this selector was for style tags in the JSX pragma? That's why it attempts to get it from parent of the actual element?

Reply:
its for the SSR case/local dev

```
<StyledDiv css={{ color: 'blue' }}>
```

ends up making

```
<style>.a
<style>.b
<div class="a b">
```

does this work?

Reply:
Hey man, not sure if the matcher would work in server mode currently. Because currently we're looking at elements rather than html strings. Am i missing something?

Reply:
ah well, yeah, true

all good

Reply:
Probably wouldn't be too hard to get working in the future though. was just playing around with it then :)